conf.py

```diff
@@ -352,8 +352,14 @@ def xapian_run_example_command(ex):
         return "%s -unsafe -target:exe -out:%s.exe %s -r:XapianSharp.dll\n./%s.exe" \
             % (csc, ex, xapian_code_example_filename(ex), ex)
     elif highlight_language == 'java':
-        javac = get_tool_name('JAVAC', 'javac')
-        java = get_tool_name('JAVA', 'java')
+        java_bindings_dir = os.environ.get("JAVA_BINDINGS_DIR")
+        classpath = ''
+        java_library_path = ''
+        if java_bindings_dir is not None:
+            classpath = ' -classpath ' + java_bindings_dir + 'xapian_jni.jar:./code/java/'
```

Comment:
I'm not sure that `os.pathsep` is guaranteed to be correct (the OS and Java could conceivably have different ideas on this), but it seems a better plan than hard coding `:`.

You can ask Java what it is, but I'm not sure how to without compiling and running a small program, such as:

```
public class conftest { public static void main(String[] args) { System.out.println(System.getProperty("path.separator")); } }
```

That's what we do in `xapian-bindings/configure.ac` to probe a value for `JAVA_PATHSEP`.